Check for 'eap' processes in JBoss scan

noahl opened the pull request:
See whether any processes on the system have 'eap' in their process
name or arguments. This will help detect JBoss EAP installations.

In addition, consolidate some duplicate functionality into a single function.

Closes #339 .

Commit:
Check for 'eap' processes in JBoss scan

See whether any processes on the system have 'eap' in their process
name or arguments. This will help detect JBoss EAP installations.

Commit:
Fix lint errors

Thanks to flake8 for the review.

chambridge commented on roles/jboss_eap/tasks/main.yml:
```diff
@@ -18,3 +18,8 @@
         - /usr/log/jboss-as
       when: '"jboss.eap.common-directories" in facts_to_collect'
+    - name: gather jboss.eap.processes
+      raw: pgrep -f eap
```
looks like a new program we need to depend on. How common is this to Linux platforms vs. `ps -f | grep eap`

chambridge replied:
do we know what Wildfly looks like here? So we can differentiate, I know Mark was looking at both product and upstream.

noahl replied:
I just did some research. Looks like pgrep has been in the procps package since at least 2002 (https://gitlab.com/procps-ng/procps/commits/master/pgrep.c), although I don't know if it's always been packaged in distributions. I'm okay switching to `ps`.

noahl replied:
I checked the latest Wildfly tar distribution. It has a very similar-looking folder hierarchy, but rooted at a folder named `wildfly-<version>` instead of `jboss-eap-<version>`.

noahl replied:
I found a file listing for the wildfly RPM. Looks like a similar directory structure to the EAP one, but again, the base directory is wildfly instead of eap7. https://www.rpmfind.net/linux/RPM/fedora/devel/rawhide/x86_64/w/wildfly-10.1.0-11.fc27.noarch.html

chambridge commented on library/spit_results.py:
```diff
@@ -288,6 +288,45 @@ def classify(key, fact_names, classifications):
+
+    if (('rc' not in raw_output or
+         'stdout_lines' not in raw_output) and 'results' not in raw_output):
+        return {this_fact: 'Error: "{0}" not run'.format(command)}, None
```
Do we want to differentiate between this and line 319 in the output. Maybe something like `Error: data from "{0}" command could not be obtained`? Might help with debugging, thoughts?

noahl replied:
I like that! Done.

Commit:
Use 'ps | grep' instead of 'pgrep'

Reduces the number of programs we depend on. Thanks to chambrid for
the review.

noahl commented:
@chambridge I just pushed a new version that I think addresses all of your comments. Please take a look.

chambridge reviewed (approved):
👍
